cli/src/check.rs

```diff
@@ -0,0 +1,197 @@
+use anyhow::Result;
+use k8s_openapi::{
+    api::{apps::v1::Deployment, core::v1::Namespace},
+    apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition,
+};
+use kube::{api::ListParams, Api, Client, ResourceExt};
+use serde::Serialize;
+
+const CHECK: &str = "√";
+const EX: &str = "×";
+
+#[derive(Serialize)]
+struct CheckOutput {
+    success: bool,
+    categories: Vec<Category>,
+}
+
+#[derive(Serialize)]
+struct Category {
+    category_name: &'static str,
+    checks: Vec<CheckResult>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "lowercase")]
+enum CheckStatus {
+    Success,
+    Error,
+}
+
+#[derive(Serialize, Default)]
+pub struct CheckResult {
+    description: &'static str,
+    result: CheckStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hint: Option<&'static str>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    error: Option<String>,
+}
```

Comment:
since, AFAICT, `hint` and `error` will only be present if `result` is `CheckStatus::Error`, I wonder if we could move them to fields on the `Error` variant, like this:
```suggestion
enum CheckStatus {
    Success,
    Error { hint: &'static str, error: String },
}

#[derive(Serialize, Default)]
pub struct CheckResult {
    description: &'static str,
    result: CheckStatus,
}
```

but, I don't know if that would serialize the way we want it to?

Reply:
yeah, I think we'd need some custom serialization to make that work.  I built the structs to reflect the desired json structure, even though it admits some invalid combinations.